packages/connectivity/src/scp-cf/destination/destination-service.ts

```diff
@@ -323,6 +328,42 @@ function errorMessageFromResponse(
     : '';
 }
 
+function retryDestination(
+  destinationName: string
+): Middleware<AxiosResponse<DestinationJson>, HttpMiddlewareContext> {
+  return options => () => {
+    let retryCount = 1;
+    return asyncRetry.default(
+      async bail => {
+        try {
+          const destination = await options.fn();
+          if (retryCount < 3) {
+            retryCount++;
+            // this will throw if the destination does not contain valid auth headers and a second try is done to get a destination with valid tokens.
+            await buildAuthorizationHeaders(parseDestination(destination.data));
+          }
+          return destination;
+        } catch (error) {
+          const status = error?.response?.status;
+          if (status.toString().startsWith('4')) {
```

Comment:
```suggestion
          if (status >= 400 && status < 500) {
```
[pp] It's slightly shorter and avoids coercing to string.